features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/MD5HashGenerator.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import com.datadog.android.api.InternalLogger
+import java.security.MessageDigest
+import java.security.NoSuchAlgorithmException
+import java.util.Locale
+
+internal class MD5HashGenerator(
+    private val logger: InternalLogger
+) : HashGenerator {
+    override fun generate(input: ByteArray): String {
```

Comment:
I would propose to make return type as `String?`. Returning empty string makes a little sense semantically (as we have to check if it is not empty later), because proper hash can never be an empty string. I understand, that this is a fallback in case of an exception, but having `null` as a fallback looks more logical to me.

Reply:
np, I'll make this change